detox/android/detox/src/full/java/com/wix/detox/DetoxManager.java

```diff
@@ -69,13 +76,17 @@ public String invoke(Throwable t) {
 
     void start() {
         if (detoxServerUrl != null && detoxSessionId != null) {
-            handler.post(new Runnable() {
```

Comment:
Was it the source of the problem? So that now I see we are using a thread factory?

Reply:
We have >1 thread, to run things (anything) concurrently